cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -489,30 +501,19 @@ public boolean apply(ProgramRuntimeService.RuntimeInfo runtimeInfo) {
   }
 
   /**
-   * Delete the metrics for an application, or if null is provided as the application ID, for all apps.
+   * Delete the metrics for an application.
    *
    * @param applicationId the application to delete metrics for.
-   * If null, metrics for all applications in the namespace are deleted.
    */
-  private void deleteMetrics(String namespaceId, String applicationId) throws Exception {
-    Collection<ApplicationSpecification> applications = Lists.newArrayList();
-    NamespaceId namespace = new NamespaceId(namespaceId);
-    if (applicationId == null) {
-      applications = this.store.getAllApplications(namespace);
-    } else {
-      ApplicationSpecification spec = this.store.getApplication(namespace.app(applicationId));
-      applications.add(spec);
-    }
-
+  private void deleteMetrics(ApplicationId applicationId) throws Exception {
```

Comment:
This function is not used. For the function `removeAll` to delete all apps, it calls `deleteApp(id, appSpec);` on every app.